test/backend/routes.spec.js

```diff
@@ -198,60 +223,192 @@ describe('Routes - ', () => {
         .catch(done);
     });
 
-    it('index page - returns 200', function(done) {
-        GET('')
-        .then(res => {
-            assert.equal(res.status, 200);
-            done();
-        })
+    it('allows access to login page without logging in', function(done) {
+        GET('login')
+        .then(() => done())
         .catch(done);
     });
 
-    // Settings
-    it('settings/urls - returns 200 and the urls', function(done) {
-        GET('settings/urls')
-        .then(res => res.json().then(json => {
-            assert.equal(res.status, 200);
-            assert.equal(json.http, 'http://localhost:9494');
-            assert.equal(json.https, '');
-            done();
-        }))
+    it('allows access to connections page without logging in', function(done) {
+        GET('')
+        .then(() => done())
         .catch(done);
     });
 
-    it('GET /settings returns some of the settings', function(done) {
-        saveSetting(
-            'USERS',
-            [{'username': 'chris', 'accessToken': 'lahlahlemons'}]
-        );
+    it('does not allow access to settings when not logged in', function(done) {
         GET('settings')
         .then(res => res.json().then(json => {
+            assert.equal(res.status, 401);
             assert.deepEqual(json, {
-                'PLOTLY_URL': 'https://plot.ly',
-                'USERS': ['chris']
+                "error": {"message": "Please login to access this page."}
             });
             done();
         })).catch(done);
     });
 
-    it('PATCH /settings sets some settings', function(done) {
-        const newSettings = {
-            'PLOTLY_API_DOMAIN': 'acme.plot.ly',
-            'PLOTLY_API_SSL_ENABLED': false
-        };
-        PATCH('settings', newSettings).then(res => res.json().then(json => {
-            assert.equal(
-                getSetting('PLOTLY_API_SSL_ENABLED'),
-                newSettings.PLOTLY_API_SSL_ENABLED
-            );
-            assert.equal(
-                getSetting('PLOTLY_API_DOMAIN'),
-                newSettings.PLOTLY_API_DOMAIN
+    it('Oauth fails when user not present in ALLOWED_USERS', function(done) {
+
+        // empty allowed_users list
+        saveSetting('ALLOWED_USERS', []);
+
+        POST('oauth2', {access_token: accessToken})
+        .then(res => res.json().then(json => {
+            assert.equal(res.status, 403);
+            assert.deepEqual(json,
+              {error: {message: `User ${username} is not allowed to view this app`}}
             );
             done();
         })).catch(done);
     });
 
+    it('allows access to settings when logged in and present in ALLOWED_USERS', function(done) {
+        saveSetting('ALLOWED_USERS', [username]);
+
+        POST('oauth2', {access_token: accessToken})
+        .then(res => {
+            GET('settings')
+            .then(res => {
+                assert.equal(res.status, 200);
+                done();
+            }).catch(done);
```

Comment:
if you `return` this `GET` then i believe that you can remove this `.catch(done)` (as long as you add it to the very end)